src/library/scala/runtime/ClassValueCompat.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.runtime
+
+private[scala] abstract class ClassValueCompat[T] {
+
+  private def checkClassValueAvailability(): Boolean =
+    try {
+      Class.forName("java.lang.ClassValue", false, classOf[Object].getClassLoader)
+      true
+    } catch {
+      case _: ClassNotFoundException => false
+    }
+
+  private val instance = if (checkClassValueAvailability()) new JavaClassValue() else new FallbackClassValue()
+
+  private class JavaClassValue extends ClassValue[T] with ClassValueInterface {
+    override def computeValue(cls: Class[_]): T = ClassValueCompat.this.computeValue(cls)
+  }
+
+  private class FallbackClassValue extends ClassValueInterface {
+    override def get(cls: Class[_]): T = ClassValueCompat.this.get(cls)
```

Comment:
If `java.lang.ClassValue` isn't present, doesn't `ClassValueCompat#get` result in an endless loop between `ClassValueCompat#get` and `FallbackClassValue#get`?

Reply:
oops, yes it has to be `computeValue`, got it fixed, thank you